lib/msal-common/src/response/AuthenticationResult.ts

```diff
@@ -27,7 +26,7 @@ export type AuthenticationResult = {
     scopes: Array<string>;
     account: AccountInfo;
     idToken: string;
-    idTokenClaims: StringDict;
+    idTokenClaims: Object;
```

Comment:
We have an IdTokenClaims type, can we use that instead of the generic "Object"?

Reply:
We decided to go with a generic Object given that ID Token Claims can conceivably contain claims of any type (excluding functions). The IdTokenClaims type wouldn't solve this particular issue because it defines all the listed claims as string-typed.

Reply:
In that case, can we update the typing in `IdTokenClaims`? Each specific claim should in theory have their own type and we should make sure our internal type complies with what the server is sending. e.g. We can expect `exp` to always be `number`, `iss` to always be `string`, etc.

Reply:
We could attempt to explicitly type all of the possible claims, but my understanding is that those may change, and there are also many optional claims that applications can enable. Ultimately, we want to fix bugs like #2084 and #1766 where we have incorrectly asserted at all claims will be strings. And it would be easier and future-proof to only guarantee that it will be an object, instead of attempting to document all possible claims.

Reply:
This value of this is set to an object that is type `IdTokenClaims` in `ResponseHandler` and the `IdToken` constructor so I'm not sure changing it from `StringDict` to `Object` here is really going to solve this problem on it's own. We would either need to change the types in all places in the code or update the type for each claim.

Reply:
I agree that only changing it here doesn't solve the problem. I was initially unaware that the IdTokenClaims type existed in the library, so I only knew to update the AuthenticationResult type that was throwing the error in #2084.  So if we were to decide on a generic Object type to replace IdTokenClaims, I'd apply that change in the ResponseHandler and the IdToken constructor (and anywhere else the IdTokenClaims type is found).

Having said that, is there any value in explicitly typing each claim? I don't think I have enough context to foresee what kind of future bug may arise for having the claims ambiguously typed as generic Object attributes (maybe the fact that developers won't immediately know which claims are supported by default?)